apps/api/src/domain/models/bibliographic-reference/court-case-bibliographic-reference/commands/create-court-case-bibliographic-reference.command.integration.spec.ts

```diff
@@ -0,0 +1,185 @@
+        dateDecided: '2020-4-23',
+        court: 'BC Supreme Court',
+        url: 'https://bcsupremecourt.com',
+        pages: '47-78',
+    },
+});
+
+const dummyFSAFactory = new DummyCommandFSAFactory(buildValidCommandFSA);
+
+describe(`The command: ${commandType}`, () => {
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let commandHandlerService: CommandHandlerService;
+
+    let idManager: IIdManager;
+
+    let assertionHelperDependencies: CommandAssertionDependencies;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, idManager } = await setUpIntegrationTest({
+            ARANGO_DB_NAME: testDatabaseName,
+        }));
+
+        assertionHelperDependencies = {
+            testRepositoryProvider,
+            commandHandlerService,
+            idManager,
+        };
+
+        commandHandlerService.registerHandler(
+            commandType,
+            new CreateCourtCaseBibliographicReferenceCommandHandler(
+                testRepositoryProvider,
+                idManager
+            )
+        );
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    afterEach(async () => {
+        await testRepositoryProvider.testTeardown();
+    });
+
+    describe('when the command is valid', () => {
+        it('should succeed with appropriate updates to the database', async () => {
+            await assertCreateCommandSuccess(assertionHelperDependencies, {
+                buildValidCommandFSA,
+                initialState,
+                systemUserId: dummySystemUserId,
+                checkStateOnSuccess: async ({ id }: CreateCourtCaseBibliographicReference) => {
+                    await assertResourcePersistedProperly(idManager, testRepositoryProvider, {
+                        id,
+                        type: ResourceType.bibliographicReference,
+                    });
+                },
+            });
+        });
+    });
+
+    describe('when the command is invalid', () => {
+        describe('when the payload has an invalid type', () => {
+            generateCommandFuzzTestCases(CreateCourtCaseBibliographicReference).forEach(
+                ({ description, propertyName, invalidValue }) => {
+                    describe(`when the property: ${propertyName} has the invalid value: ${invalidValue} ${description}`, () => {
+                        it('should fail with the appropriate error', async () => {
+                            await assertCommandFailsDueToTypeError(
+                                assertionHelperDependencies,
+                                { propertyName, invalidValue },
+                                buildValidCommandFSA('unused-id')
+                            );
+                        });
+                    });
+                }
+            );
+        });
+    });
+
+    /**
+     * TODO Ideally, we would remove duplication with other CreateBibliographicReference
```

Comment:
Let's add these specific files to the sonar cloud ignore for now. It rightfully complains about duplication, but we're not entirely sure this code will survive the scrutiny of our upcoming round of re-modelling.